ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/AudienceBasedTokenAuthIT.scala

```diff
@@ -0,0 +1,120 @@
+
+import com.daml.error.ErrorsAssertions
+import com.daml.ledger.api.auth.{StandardJWTPayload, StandardJWTTokenFormat}
+import com.daml.ledger.api.v1.admin.package_management_service._
+import com.daml.ledger.runner.common.Config
+import com.daml.ledger.sandbox.SandboxOnXForTest.singleParticipant
+import com.daml.platform.config.JwtAudience
+import com.daml.platform.sandbox.auth.AudienceBasedTokenAuthIT.ExpectedAudience
+import com.daml.test.evidence.scalatest.ScalaTestSupport.Implicits._
+
+import java.io.File
+import java.time.{Duration, Instant}
+import scala.concurrent.Future
+
+class AudienceBasedTokenAuthIT extends ServiceCallAuthTests with ErrorsAssertions {
+  override def config: Config = super.config.copy(
+    participants = singleParticipant(
+      jwtAudience = JwtAudience(enabled = true, audience = Some(ExpectedAudience))
+    )
+  )
+
+  override def expectsAudienceBasedTokens: Boolean = true
+
+  override def serviceCallName: String = ""
+
+  override def packageFiles: List[File] = List.empty
+
+  override protected def serviceCall(context: ServiceCallContext): Future[Any] =
+    stub(PackageManagementServiceGrpc.stub(channel), context.token)
+      .listKnownPackages(ListKnownPackagesRequest())
+
+  private def toContext(payload: StandardJWTPayload): ServiceCallContext = ServiceCallContext(
+    Some(
+      toHeader(
+        payload,
+        audienceBasedToken = true,
+      )
+    )
+  )
+
+  val expectedAudienceToken: StandardJWTPayload = StandardJWTPayload(
+    issuer = None,
+    participantId = None,
+    userId = "participant_admin",
+    exp = None,
+    format = StandardJWTTokenFormat.ParticipantId,
+    audiences = List(ExpectedAudience),
+  )
+
+  val multipleAudienceWithExpectedToken: StandardJWTPayload =
+    expectedAudienceToken.copy(audiences = List(ExpectedAudience, "additionalAud"))
+
+  val noAudienceWithExpectedToken: StandardJWTPayload =
+    expectedAudienceToken.copy(audiences = List("aud1", "aud2"))
+
+  val expiredToken: StandardJWTPayload =
+    expectedAudienceToken.copy(exp = Some(Instant.now().plusNanos(Duration.ofDays(-1).toNanos)))
+
+  val unknownUserToken: StandardJWTPayload =
+    expectedAudienceToken.copy(userId = "unknown_user")
+
+  it should "allow access to an endpoint with the token which is matching intended audience" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with a JWT with intended audience"
+    ) in {
+    expectSuccess {
+      serviceCall(toContext(expectedAudienceToken))
+    }
+  }
+
+  it should "allow access to an endpoint with the token with multiple audiences which is matching expected audience" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with a JWT with intended audience"
+    ) in {
+    expectSuccess {
+      serviceCall(toContext(multipleAudienceWithExpectedToken))
+    }
+  }
+
+  it should "deny access with no intended audience" taggedAs securityAsset.setAttack(
```

Comment:
Actually the test has audience rather but it is the incorrect one. I would add another test where the audience is really missing